File: src/main/java/com/lessmarkup/dataobjects/MultiFactorAuthorization.java
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
package com.lessmarkup.dataobjects;

/**
 *
 * @author User
 */
public enum MultiFactorAuthorization {
    NONE(0),
    ALWAYS(1),
    CHANGED_IP(2);
    
    private final int value;
    
    MultiFactorAuthorization(int value) {
        this.value = value;
    }
    
    public static MultiFactorAuthorization of(int value) {
        switch (value) {
            case 0:
                return NONE;
            case 1:
                return ALWAYS;
            case 2:
                return CHANGED_IP;
            default:
                return null;
        }
    }
    
    public int getValue() {
        return this.value;
    }
}

File: src/main/scala/com/lessmarkup/dataobjects/MultiFactorAuthorization.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.dataobjects

object MultiFactorAuthorization {

  def apply(value: Int) = new MultiFactorAuthorization(value)

  final val NONE = MultiFactorAuthorization(0)
  final val ALWAYS = MultiFactorAuthorization(1)
  final val CHANGED_IP = MultiFactorAuthorization(2)
}

final class MultiFactorAuthorization(value: Int)
